server/src/tree-sitter/analyzer.ts

```diff
@@ -882,23 +882,53 @@ export default class Analyzer {
   }
 
   /**
- * Resolve the symbol if it contains variable expansion syntax in its overrides
- */
-  public resolveSymbol (symbol: BitbakeSymbolInformation, lookUpSymbolList: BitbakeSymbolInformation[]): BitbakeSymbolInformation {
-    const overridesToResolve = symbol.overrides.filter((override) => typeof override !== 'string').map((override) => (override as { variableName: string, value?: string }).variableName)
+   *
+   * @param symbol A symbol that contains variable expansion syntax. e.g. VAR:${PN} or require ${PN}.inc
+   * @param lookUpSymbolList A list of symbols to look up the final value of the variables in the variable expansion syntax
+   *
+   * Resolve the symbol if it contains variable expansion syntax in its overrides
+   */
+  public resolveSymbol (symbol: BitbakeSymbolInformation | string, lookUpSymbolList: BitbakeSymbolInformation[]): BitbakeSymbolInformation | string {
+    const variableToResolve: string[] = []
 
-    const resolvedOverrdies: string[] = []
-    overridesToResolve.forEach((override) => {
-      const value = lookUpSymbolList.find((symbol) => symbol.name === override)?.finalValue
-      value !== undefined && resolvedOverrdies.push(value)
-    })
+    if (typeof symbol !== 'string') {
+      variableToResolve.push(...symbol.overrides.filter((override) => typeof override !== 'string').map((override) => (override as { variableName: string, value?: string }).variableName))
```

Comment:
You could avoid doing a type assertion like this
```suggestion
      const variableToResolve = symbol.overrides.reduce<string[]>((acc, override) => {
        if (typeof override !== 'string') {
          acc.push(override.variableName)
        }
        return acc
      }, [])
```